File: spark-study-java/src/main/java/com/zhouq/spark/sql/ParquetLoadData.java
package com.zhouq.spark.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

import java.util.List;

/**
 * Parquet 数据源加载之编程方式加载数据
 * Create by zhouq on 2019/9/9
 */
public class ParquetLoadData {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("ParquetLoadData").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        SQLContext sqlContext = new SQLContext(sc);

        // 读取一个 parquet 文件，创建一个DataFrame
        DataFrame usersDF = sqlContext.read().load("H:\\bigdatatest\\spark\\sql\\users.parquet");

        // 将DataFrame 注册为一个临时表，然后使用SQL 查询需要的数据
        usersDF.registerTempTable("users");
        DataFrame nameFromUsersDF = sqlContext.sql("select name from users");

        // 对查询出来的DataFrame 进行 transformation 操作，处理数据，然后打印出来
        List<String> userNames = nameFromUsersDF.javaRDD().map(new Function<Row, String>() {
            @Override
            public String call(Row row) throws Exception {
                return "name:" + row.getString(0);
            }
        }).collect();

        for (String userName : userNames) {
            System.out.println(userName);
        }
        sc.stop();
    }
}

File: spark-study-scala/src/main/java/com/zhouq/spark/sql/ParquetLoadData.scala
package com.zhouq.spark.sql

import org.apache.spark.sql.{DataFrame, SQLContext}
import org.apache.spark.{SparkConf, SparkContext}

/**
  * Create by zhouq on 2019/9/9
  *
  */
object ParquetLoadData {
  def main(args: Array[String]): Unit = {
    val conf: SparkConf = new SparkConf().setMaster("local").setAppName("ParquetLoadData")
    val sc = new SparkContext(conf)

    val sqlContext = new SQLContext(sc)

    val usersDF: DataFrame = sqlContext.read.load("H:\\bigdatatest\\spark\\sql\\users.parquet")

    usersDF.registerTempTable("users")

    val userNamesDF: DataFrame = sqlContext.sql("select name from users")

    userNamesDF.rdd.map(row => ("name:" + row.getString(0))).collect().foreach(println)

    sc.stop()
  }

}
